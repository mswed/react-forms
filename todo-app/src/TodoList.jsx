import { useState } from 'react';
import NewTodoForm from './NewTodoForm';
import Todo from './Todo';

const TodoList = () => {
  const [todoList, setTodoList] = useState([]);
  const newTodo = (todo) => setTodoList((todos) => [...todos, todo]);
  const removeTodo = (id) => {
    setTodoList(todoList.filter((todo) => todo.id !== id));
  };

  return (
    <div>
      <h1>My Todos:</h1>
      <NewTodoForm newTodo={newTodo} />
      {todoList.map((todo) => (
        <Todo text={todo.text} id={todo.id} key={todo.id} remove={removeTodo} />
      ))}
    </div>
  );
};

export default TodoList;
