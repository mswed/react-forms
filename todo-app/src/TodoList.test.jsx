import { it, expect } from 'vitest';
import { render, fireEvent } from '@testing-library/react';
import TodoList from './TodoList';

it('Renders without crashing', async () => {
  render(<TodoList />);
});

it('Matches the snapshot', async () => {
  const { asFragment } = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it('Adds a todo', async () => {
  const todos = render(<TodoList />);
  const newTodoInput = todos.getByLabelText('Todo Text:');
  const newTodoBtn = todos.queryByText('Create!');

  // Expect the form to be empty
  expect(todos.queryByText('X')).not.toBeInTheDocument();

  fireEvent.change(newTodoInput, { target: { value: 'upload to git' } });
  fireEvent.click(newTodoBtn);

  expect(todos.queryByText('upload to git')).toBeInTheDocument();
});
