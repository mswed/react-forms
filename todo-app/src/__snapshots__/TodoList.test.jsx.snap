// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      My Todos:
    </h1>
    <form>
      <h2>
        New Todo
      </h2>
      <label
        for="text"
      >
        Todo Text:
      </label>
      <input
        id="text"
        name="text"
        placeholder="Todo text"
        type="text"
        value=""
      />
      <button>
        Create!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
